decidim-core/app/controllers/decidim/direct_uploads_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Decidim
+  class DirectUploadsController < ActiveStorage::DirectUploadsController
+    include Decidim::NeedsOrganization
+    skip_before_action :verify_organization
+
+    before_action :check_organization!
+    before_action :check_authenticated!
+    before_action :validate_direct_upload
+
+    protected
+
+    def verify_organization; end
+
+    def validate_direct_upload
+      maximum_allowed_size = current_organization.settings.upload_maximum_file_size
+
+      extension = File.extname(blob_args[:filename]).delete(".")
+
+      Rails.logger.debug content_types.inspect
+
+      head :unprocessable_entity unless maximum_allowed_size.try(:to_i) >= blob_args[:byte_size].try(:to_i)
+      head :unprocessable_entity unless content_types.any? { |pattern| pattern.match?(blob_args[:content_type]) }
+      head :unprocessable_entity unless content_types.any? { |pattern| pattern.match?(MiniMime.lookup_by_extension(extension).content_type) }
```

Comment:
```suggestion
      head :unprocessable_entity unless content_types.any? { |pattern| pattern.match?(MiniMime.lookup_by_extension(extension)&.content_type) }
```

One more nil-safety issue here, since `MiniMime.lookup_by_extension("...")` will return `nil` on undetected file extensions. `Regexp#match?` seems to accept `nil` but `nil` does not respond to `content_type`.